sql: document RLS DDL statements

spilchen opened the pull request:
RLS-related DDL was introduced in v25.1 but left undocumented while the feature was still under development. With RLS approaching completion, this commit adds documentation for the relevant DDL to ensure it is included in the 25.2 release docs.

Informs: #137122
Epic: CRDB-11724

Release note (sql): Added documentation for previously hidden DDL statements related to row-level security (RLS). RLS enables fine-grained access control by enforcing policies that determine which rows a user can read or modify.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/143510)
<!-- Reviewable:end -->

Dedej-Bergin reviewed (commented):
Nice changes!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/143510#-:-OMJ6OUvB5cEqDI6EZS5:bibb57g)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`docs/generated/sql/bnf/stmt_block.bnf` line 2471 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/143510#-OMJ6Tp1-7LmtZe_BQ8A:-OMJ6Tp1-7LmtZe_BQ8B:byv20lu) ([raw file](https://github.com/cockroachdb/cockroach/blob/485346d28306aa7cd78f3a71c431cae41b7907a6/docs/generated/sql/bnf/stmt_block.bnf#L2471)):*
> ```
> 	| 'WITH' 'CHECK' '(' a_expr ')'
> 	| 'USING' '(' a_expr ')'
> 	| 
> ```

Why do we have this extra `|` here? I see in other examples that we don't have it.

Example from line 678:
```
drop_ddl_stmt ::=
	drop_database_stmt
	| drop_index_stmt
	| drop_table_stmt
	| drop_view_stmt
	| drop_sequence_stmt
	| drop_schema_stmt
	| drop_type_stmt
	| drop_func_stmt
	| drop_proc_stmt
	| drop_trigger_stmt
	| drop_policy_stmt
```


<!-- Sent from Reviewable.io -->

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/143510)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @Dedej-Bergin)

___
*[`docs/generated/sql/bnf/stmt_block.bnf` line 2471 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/143510#-OMJ6Tp1-7LmtZe_BQ8A:-OMMJGQM1QZaqq0W4gjb:b-gpofj0) ([raw file](https://github.com/cockroachdb/cockroach/blob/485346d28306aa7cd78f3a71c431cae41b7907a6/docs/generated/sql/bnf/stmt_block.bnf#L2471)):*
<details><summary><i>Previously, Dedej-Bergin (Bergin Dedej) wrote…</i></summary><blockquote>

Why do we have this extra `|` here? I see in other examples that we don't have it.

Example from line 678:
```
drop_ddl_stmt ::=
	drop_database_stmt
	| drop_index_stmt
	| drop_table_stmt
	| drop_view_stmt
	| drop_sequence_stmt
	| drop_schema_stmt
	| drop_type_stmt
	| drop_func_stmt
	| drop_proc_stmt
	| drop_trigger_stmt
	| drop_policy_stmt
```
</blockquote></details>

It's the empty case. It allows you to define a policy without any expressions. For example, this is valid syntax:
```
CREATE POLICY p ON t;
```


<!-- Sent from Reviewable.io -->

spilchen committed:
sql: document RLS DDL statements

RLS-related DDL was introduced in v25.1 but left undocumented while the
feature was still under development. With RLS approaching completion,
this commit adds documentation for the relevant DDL to ensure it is
included in the 25.2 release docs.

Informs: #137122
Epic: CRDB-11724

Release note (sql): Added documentation for previously hidden DDL
statements related to row-level security (RLS). RLS enables fine-grained
access control by enforcing policies that determine which rows a user
can read or modify.

rafiss reviewed (approved):
> Reviewed 15 of 15 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/143510)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @Dedej-Bergin)



<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR!

bors r+

spilchen commented:
blathers auth-check

spilchen commented:
You shall pass
